File: 프로그래머스/2/42842. 카펫/카펫.kt
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        var answer = intArrayOf()
        
        var col = (brown - 2) / 2 - 2
        var row = 1
        
        var ans = 0
        
        while(ans != yellow){
            ans = col * row
            
            if(ans != yellow){
                col--
                row++
            }
        }
        
        answer += col + 2
        answer += row + 2
        
        
        return answer
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int col = (brown - 2) / 2 - 2;
    int row = 1;
    
    int sol = 0;
    
    while(sol != yellow){
        sol = col * row;
        
        if(sol != yellow){
            col--;
            row++;
        }
    }
    
    answer.push_back(col+2);
    answer.push_back(row+2);
    
    return answer;
}